### PR TITLE
Enable Link-Time Optimization (LTO)

### LINKED ISSUE #1: Enable Link-Time Optimization (LTO)
Hi!

I noticed that in the `Cargo.toml` file Link-Time Optimization (LTO) for the project is not enabled. I suggest switching it on since it will reduce the binary size (always a good thing to have) and will likely improve the application's performance a bit. If you want to read a bit more about LTO, I can recommend starting from [this](https://doc.rust-lang.org/rustc/codegen-options/index.html#lto) Rustc documentation.

I suggest enabling LTO only for the Release builds so as not to sacrifice the developers' experience while working on the project since LTO consumes an additional amount of time to finish the compilation routine. If you think that a regular Release build should not be affected by such a change as well, then I suggest adding an additional `dist` or `release-lto` profile where in addition to regular `release` optimizations LTO will also be added. Such a change simplifies life for maintainers and others interested in the project persons who want to build the most performant version of the application. If we enable it on the Cargo profile level, users, who install the application with `cargo install` will get the LTO-optimized version of the game "automatically". E.g., check `cargo-outdated` Release [profile](https://github.com/kbknapp/cargo-outdated/blob/master/Cargo.toml#L48). You also could be interested in other optimization options like `codegen-units = 1`, etc.

Basically, it can be enabled with the following lines:

```
[profile.release]
lto = true
```

I have made quick tests (Fedora 41, Rust 1.82) by adding `lto = true` to the Release profile. The binary size reduction is from 13 Mib to 9.8 Mib.

Thank you.

### PR DESCRIPTION
**Pull request by zamazan4ik**
Resolves https://github.com/0xb-s/ssh-browser/issues/1

### CODE REVIEW

**Comment by 0xb-s** [comment]
Thank you! Just a few minor nits.

File: Cargo.toml [hunk 1]
```diff
@@ -12,3 +12,6 @@ rfd = "0.13"
 libc = "0.2.168"
 serde = { version = "1.0.216", features = ["derive"] }
 serde_json = "1.0.133"
+
```
Comment (0xb-s) [inline Cargo.toml:15]: Could you please add a small comment explaining what this option does and include a link to the relevant documentation for better understanding?

Comment (zamazan4ik) [reply Cargo.toml:15]: Sure. Just did it

**Comment by 0xb-s** [comment]
LGTM!

**Pull request merged**